include/cvc5/c/cvc5.h

```diff
@@ -0,0 +1,3936 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Aina Niemetz
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2023 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The cvc5 C API.
+ */
+
+#ifndef CVC5__C_API__CVC5_H
+#define CVC5__C_API__CVC5_H
+
+#if __cplusplus
+extern "C" {
+#endif
+
+#define CVC5_API_USE_C_ENUMS
+#include <cvc5/cvc5_kind.h>
+#include <cvc5/cvc5_sort_kind.h>
+#include <cvc5/cvc5_types.h>
+#undef CVC5_API_USE_C_ENUMS
+
+#include <stdint.h>
+#include <stddef.h>
+
+/* -------------------------------------------------------------------------- */
+
+/**
+ * Encapsulation of a three-valued solver result, with explanations.
+ */
+typedef struct Cvc5Result Cvc5Result;
+
+/**
+ * Encapsulation of a solver synth result.
+ *
+ * This is the return value of the API functions:
+ *   - cvc5_check_synth()
+ *   - cvc5_check_synth_next()
+ *
+ * which we call "synthesis queries".  This class indicates whether the
+ * synthesis query has a solution, has no solution, or is unknown.
+ */
+typedef struct Cvc5SynthResult Cvc5SynthResult;
+
+/**
+ * The sort of a cvc5 term.
+ */
+typedef uint64_t Cvc5Sort;
```

Comment:
How did you decide what to make a struct and what to make a uint64_t?  I guess uint64_t is for shared pointers?  How do you handle the reference counting?

Reply:
As for what to make a struct vs uint64_t: Everything that we have a unique id for internally, is a uint64_t. The rest is pointers. We will either have to go the static thread_local route below for these, or allocate memory and add `_delete` functions to free the allocated memory for these objects.

This is actually related to the question about reference counting. As a first step,  would suggest to not introduce external reference counting on the C API, i.e., no garbage collection. When it is stable, we would add reference counting and a `_copy` and `_release` for users to be able to manually release and copy terms and sorts.

Reply:
Sounds good